### ISSUE TITLE
NCEPLIB-bufr crashes on WMO-formated BUFR file for Radar VAD-winds.

### ISSUE THREAD

**Issue by rmclaren** [NOAA-EMC/NCEPLIBS-bufr#187]
Greg Thompson reported an issue with a WMO BUFR file ([test.BUFR.gz](https://github.com/NOAA-EMC/NCEPLIBS-bufr/files/7997146/test.BUFR.gz)). According to Greg the data in the file is "WMO-format BUFR file of Radar VAD-winds".

I tried it and found that the problem happens when you load the file and then attempt to call `ireadmg` the first time. You get the following error message:

```
 ***********BUFR ARCHIVE LIBRARY ABORT**************
 BUFRLIB: STSEQ - NOT ENOUGH REMAINING CHILD DESCRIPTORS TO COMPLETE REPLICATION FOR 103000
 ***********BUFR ARCHIVE LIBRARY ABORT**************
```

Here is the sample fortran code I used to reproduce the problem:
```
        integer, parameter :: lunit = 12
        integer, parameter :: table1Unit = 13
        integer, parameter :: table2Unit = 14

        integer :: ireadmg, ireadsb
        character(8) :: subset
        integer(kind=8) :: my_idate
        integer :: lun, il, im
        integer :: iret

        open(lunit, file="/some_path/test.BUFR")
        call openbf(lunit, "SEC3", lunit)
        call mtinfo("/nceplib_bufr_install_directory/tables/", table1Unit, table2Unit)

        do while (ireadmg(lunit, lun, subset, my_idate) == 0)      !  >>>> Throws error message here
          do while (ireadsb(lunit) == 0)
            print *, "Do something here"
          end do
        end d

        call closbf(lunit)
        close(lunit)
```

**Comment by jbathegit** [comment]
The problem here is in the BUFR file itself - when unzipped it's only 76 bytes long and contains a malformed Section 3, which you can see if you use `debufr -b`:

```
Found BUFR message #      1

        Message length:          76
      Section 0 length:           8
          BUFR edition:           2

      Section 1 length:          18
          Master table:           0
    Originating center:       65535
 Originating subcenter:          -1
 Update sequence numbr:           0
    Section 2 present?: No
         Data category:           6
     Local subcategory:           0
 Internatl subcategory:          -1
  Master table version:          11
   Local table version:           4
                  Year:        2021
                 Month:           7
                   Day:          31
                  Hour:          18
                Minute:           3
                Second:          -1

 Number of data subsets:           1
     Data are observed?: Yes
   Data are compressed?: No
  Number of descriptors:           8
        1: 301001
        2: 301011
        3: 301012
        4: 301022
        5: 002003
        6: 033002
        7: 103000
        8: 031001

End of BUFR message #      1
```
Specifically, the 7th descriptor `103000` is a delayed-replication descriptor which says there are 3 more descriptors following beyond the `031001`, which itself is the indicator for 8-bit delayed-replication (i.e. up to 255 replications).  But as you can see, there are no more descriptors following the `031001`, which in turn is what the error message is pointing out.

So bottom line, there's no problem with the library; rather, the BUFR data file itself is erroneous.
